scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -0,0 +1,154 @@
+    if (isExempted) {
+      println("This file is exempted from having a test file. Hence No coverage!")
+      return mutableListOf()
+    }
+
+    val testFilePaths = findTestFile(repoRoot, filePath)
+    val testTargets = bazelClient.retrieveBazelTargets(testFilePaths)
+
+    for (testTarget in testTargets) {
+      val coverageData = RunCoverageForTestTarget(
+        rootDirectory,
+        testTarget.substringBeforeLast(".kt"),
+        commandExecutor,
+        scriptBgDispatcher
+      ).runCoverage()!!
+      coverageDataList.add(coverageData)
+    }
+    println("Coverage Data List: $coverageDataList")
+    return coverageDataList
+  }
+
+  /**
+   * Finds potential test file paths corresponding to a given source file path within a repository.
+   *
+   * @param repoRoot the root directory of the repository
+   * @param filePath The file path of the source file for which the test files are to be found.
+   * @return A list of potential test file paths that exist in the repository.
+   */
+  fun findTestFile(repoRoot: String, filePath: String): List<String> {
+    val file = File(filePath)
+    val parts = file.parent.split(File.separator)
+    val testFiles = mutableListOf<String>()
+
+    if (parts.isNotEmpty() && parts[0] == "scripts") {
+      val testFilePath = filePath.replace("/java/", "/javatests/").replace(".kt", "Test.kt")
+      if (File(repoRoot, testFilePath).exists()) {
+        testFiles.add(testFilePath)
+      }
+    } else if (parts.isNotEmpty() && parts[0] == "app") {
+      val sharedTestFilePath = filePath.replace("/main/", "/sharedTest/").replace(".kt", "Test.kt")
+      val testFilePath = filePath.replace("/main/", "/test/").replace(".kt", "Test.kt")
+      val localTestFilePath = filePath.replace("/main/", "/test/").replace(".kt", "LocalTest.kt")
+
+      if (File(repoRoot, sharedTestFilePath).exists()) {
+        testFiles.add(sharedTestFilePath)
+      }
+      if (File(repoRoot, testFilePath).exists()) {
+        testFiles.add(testFilePath)
+      }
+      if (File(repoRoot, localTestFilePath).exists()) {
+        testFiles.add(localTestFilePath)
+      }
+    } else {
+      val defaultTestFilePath = filePath.replace("/main/", "/test/").replace(".kt", "Test.kt")
+      if (File(repoRoot, defaultTestFilePath).exists()) {
+        testFiles.add(defaultTestFilePath)
+      }
+    }
+    return testFiles
+  }
+
+  private fun loadTestFileExemptionsProto(testFileExemptiontextProto: String): TestFileExemptions {
+    val protoBinaryFile = File("$testFileExemptiontextProto.pb")
+    val builder = TestFileExemptions.getDefaultInstance().newBuilderForType()
+
+    // This cast is type-safe since proto guarantees type consistency from mergeFrom(),
+    // and this method is bounded by the generic type T.
+    @Suppress("UNCHECKED_CAST")
+    val protoObj: TestFileExemptions =
+      FileInputStream(protoBinaryFile).use {
+        builder.mergeFrom(it)
+      }.build() as TestFileExemptions
+    return protoObj
```

Comment:
```suggestion
    return File("$testFileExemptiontextProto.pb").inputStream().use { stream ->
      TestFileExemptions.newBuilder().also { builder ->
        builder.mergeFrom(stream)
      }.build()
    }
```

Can be simplified a bunch, and no unsafe cast is needed.

Reply:
Okay, I just duplicated them from the existing implementations. Now they are updated as suggested. Thanks!